modules/tensor_mechanics/include/materials/ComputeThermalExpansionEigenstrainBeamBase.h

```diff
@@ -0,0 +1,55 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#ifndef COMPUTETHERMALEXPANSIONEIGENSTRAINBEAMBASE_H
+#define COMPUTETHERMALEXPANSIONEIGENSTRAINBEAMBASE_H
+
+#include "ComputeEigenstrainBeamBase.h"
+#include "DerivativeMaterialInterface.h"
+
+class ComputeThermalExpansionEigenstrainBeamBase;
+
+template <>
+InputParameters validParams<ComputeThermalExpansionEigenstrainBeamBase>();
+
+/**
+ * ComputeThermalExpansionEigenstrainBeamBase is a base class for all models that
+ * compute beam eigenstrains due to thermal expansion of a material.
+ */
+class ComputeThermalExpansionEigenstrainBeamBase
+  : public DerivativeMaterialInterface<ComputeEigenstrainBeamBase>
+{
+public:
+  ComputeThermalExpansionEigenstrainBeamBase(const InputParameters & parameters);
+
+protected:
+  virtual void computeQpEigenstrain() override;
+  /*
+   * Compute the total thermal strain relative to the stress-free temperature at
+   * the current temperature, as well as the current instantaneous thermal
+   * expansion coefficient.
+   * param thermal_strain    The current total linear thermal strain
+   *                         (\delta L / L)
+   * param instantaneous_cte The current instantaneous coefficient of thermal
```

Comment:
I don't think there's any need for `instantaneous_cte`. It's there in the solid version of this for computing fracture integrals, but that's not applicable for beams.